src/Gui/DlgExpressionInput.cpp

```diff
@@ -633,4 +581,135 @@ void ProxyWidget::paintEvent(QPaintEvent *)
+    QRgb userTextColor = 0xFF000000 | (hGrp->GetUnsigned("colorText", 
+                         0x00000000 | (defaultTextColor << 8)) >> 8);
+
+    constexpr QRgb defaultLogColor = qRgba(0, 0, 255, 255);
+    QRgb userLogColor = 0xFF000000 | (hGrp->GetUnsigned("colorLogging", 
+                        0x00000000 | (defaultLogColor << 8)) >> 8);
+
+    constexpr QRgb defaultWarningColor = qRgba(255, 170, 0, 255);
+    QRgb userWarningColor = 0xFF000000 | (hGrp->GetUnsigned("colorWarning", 
+                            0x00000000 | (defaultWarningColor << 8)) >> 8);
+
+    constexpr QRgb defaultErrorColor = qRgba(255, 0, 0, 255);
+    QRgb userErrorColor = 0xFF000000 | (hGrp->GetUnsigned("colorError", 
+                          0x00000000 | (defaultErrorColor << 8)) >> 8);
+
+    //Color priority does the heavy lifting but needs lots of arguments to evaluate every case
+    //I feel this can be factored better but for *only* 4 calls it looks good enough and fits one screen
+    this->textColorStyle = colorPriority(userTextColor, defaultTextColor, 
+                           hasStyleSheet, isDarkStyle, this->hasTextColor, this->hasTextBackgroundColor,
+                           qRgba(255, 255, 255, 255), qRgba(28, 28, 28, 179), //dark colors
+                           qRgba(0, 0, 0, 228), qRgba(255, 255, 255, 200), //light colors
+                           &_textColor, &_textBackgroundColor);
+
+    this->logColorStyle = colorPriority(userLogColor, defaultLogColor, 
+                          hasStyleSheet, isDarkStyle, this->hasLogColor, this->hasLogBackgroundColor,
+                          qRgba(96, 205, 255, 179), qRgba(28, 28, 28, 179), //dark colors
+                          qRgba(0, 95, 183, 200), qRgba(255, 255, 255, 200), //light colors
+                          &_logColor, &_logBackgroundColor);
+
+    this->warningColorStyle = colorPriority(userWarningColor, defaultWarningColor, 
+                              hasStyleSheet, isDarkStyle, this->hasWarningColor, this->hasWarningBackgroundColor,
+                              qRgba(252, 225, 0, 179), qRgba(28, 28, 28, 179), //dark colors
+                              qRgba(157, 93, 0, 200), qRgba(255, 255, 255, 200), //light colors
+                              &_warningColor, &_warningBackgroundColor);
+
+    this->errorColorStyle = colorPriority(userErrorColor, defaultErrorColor, 
+                            hasStyleSheet, isDarkStyle, this->hasErrorColor, this->hasErrorBackgroundColor,
+                            qRgba(255, 153, 164, 179), qRgba(28, 28, 28, 179), //dark colors
+                            qRgba(198, 43, 28, 200), qRgba(255, 255, 255, 200), //light colors
+                            &this->_errorColor, &this->_errorBackgroundColor);
+
+    //Old logic to decide the border and background colors of the checkboxes if there no system background is on
+    //I failed to understand excatly *why* this was needed so for now I only removed redundant variables
```

Comment:
The reason of having `noBackground` is that on some Linux system, transparent top level pop up window does not work, resulting black background instead of transparent. So the user is offer a choice to turn off `noBackground` and change to a normal dialog, in which case the checkbox style shall follow the main stylesheet.

Reply:
I see, I did read the bit about the mouse events but somehow the initial description of noBackground eluded me, I modified the comment to reflect the understanding.